src/util/PromiseUtil.ts

```diff
@@ -0,0 +1,10 @@
+// A replacement for the Promise.any() function, which is only supported
+// by >=15.0.0 NodeJs versions.
+export async function hasSome(promises: Iterable<Promise<boolean>>): Promise<boolean> {
+  for (const promise of promises) {
```

Comment:
Linear behavior though, parallel?

Reply:
I'm aware of this, I'm going to push other commits to update the hasSome implementation and to add again the cache into GroupAccessChecker (in the form suggested by @joachimvh ). Then, at the end, some squashing and rebasing will do the job of cleaning up all these commits...

Reply:
Fixed